site/content/entry/yuzu-progress-report-mar-2021/index.md

```diff
@@ -0,0 +1,143 @@
+
+Going back to kernel updates, [bunnei](https://github.com/bunnei/) [reworked and cleaned the kernel memory management code](https://github.com/yuzu-emu/yuzu/pull/6099).
+As the name implies, this is the part of the OS that allocates memory resources for games when they request them.
+With this change, the implementation matches more closely that of the Nintendo Switch.
+As always, these changes make it easier to support any future hardware revisions, along with all other kernel changes that are currently ongoing.
+
+Bunnei also fixed [a memory leak](https://github.com/yuzu-emu/yuzu/pull/6036) caused by `dummy threads`.
+These dummies are used by yuzu to interact with our emulated kernel.
+Every "real" emulated thread has a dummy associated with it.
+As explained in the previous progress report, yuzu utilizes fibers in order to emulate threads.
+However, these dummy threads don't actually use fibers.
+With this change, bunnei removed some unnecessary memory overhead by removing the creation of fibers (which would only be needed for "real" emulated threads), thus reducing the memory usage by a bit.
+
+Another bug also fixed by bunnei was related to how [JIT states are saved between page table changes](https://github.com/yuzu-emu/yuzu/pull/6100).
+A [page table](https://en.wikipedia.org/wiki/Page_table) is a scheme used to map physical memory into virtual memory, used to give processes the impression they're working with a unique, contiguous section of memory, regardless of where and how this memory is actually stored.
+Due to the way dynarmic is coded, whenever yuzu changes the page table that is currently in use, it needs to recreate the CPU JIT.
+But, until now, yuzu wasn't saving and restoring the state of this JIT properly when this happened.
+However, with this change, all the context needed by the JIT is now stored and retrieved correctly.
+
+## Input improvements
+
+Not all devices can do `N-key rollover`, or pressing as many keys/buttons at the same time as you want. 
+For this, [german77](https://github.com/german77) implemented a [toggle to hold a button for you.](https://github.com/yuzu-emu/yuzu/pull/6040)
+This way you can ask yuzu to keep pressing a button, freeing your hand and devices from it, while you move to something else! 
+To access this feature, right click the Modifier button in control settings, and select `Toggle button`.
+
+{{< imgs
+	"./toggle.png| Right click any button or modifier"
+  >}}
+
+For those using [Cemuhook](https://cemuhook.sshnuke.net/) to provide motion via an Android device, [german77](https://github.com/german77) has a nice stability improvement for you! 
+[Each individual socket connection now needs its unique client ID.](https://github.com/yuzu-emu/yuzu/pull/6004) 
+
+In a separate later PR, german77 [tests using a single UDP connection per server](https://github.com/yuzu-emu/yuzu/pull/6127) instead of per controller, reducing the error rate thanks to simplified communications.
```

Comment:
```suggestion
In a separate PR, german77 [tests using a single UDP connection per server](https://github.com/yuzu-emu/yuzu/pull/6127) instead of per controller, reducing the error rate thanks to simplified communications.
```